pkg/github/client/client.go

```diff
@@ -52,11 +53,46 @@ var runnerPerMinuteRate = map[string]float64{
 
 // New instantiates a new GitHub API client.
 func New(ctx context.Context, settings models.Settings) (*Client, error) {
-	if settings.AccessToken == "" {
-		// If access token is not set, return downstream error as it is required.
-		return nil, errorsource.DownstreamError(fmt.Errorf("access token is required"), false)
+	if settings.SelectedAuthType == "github-app" {
+		return createAppClient(settings)
 	}
 
+	if settings.SelectedAuthType == "personal-access-token" {
```

Comment:
I think we will need to handle here a case when no `SelectedAuthType` is selected and default to `personal-access-token`. Because there might be some provisioned data sources, or data sources that already exist that don't have `SelectedAuthType`.

Reply:
<img width="1500" alt="image" src="https://github.com/user-attachments/assets/4a490d14-dbb0-4a65-9ba4-518b1a8e64dc">
This error is appearing for example for my github data source that uses personal access token, but it was created in the past.

Reply:
ohhh nice find, looking into this now